### PR TITLE
single-az machinepool tc

### PR DESCRIPTION
**Pull request by amalykhi**
Singleaz test ocp-65063

the test passed after the fix
Ran 1 of 24 Specs in 27.974 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 23 Skipped
PASS

Ginkgo ran 1 suite in 28.970285431s
Test Suite Passed
[amalykhi@amalykhi-thinkpadp1gen5 terraform-provider-rhcs]$ ginkgo run --focus OCP-65071 --timeout 2h --output-dir /tmp/secret --junit-report result.xml -r --focus-file tests/e2e/. tests/e2e/..

### CODE REVIEW

File: tests/e2e/machine_pool_test.go [hunk 1]
```diff
@@ -426,5 +428,61 @@ var _ = Describe("TF Test", func() {
...
+		Context("Author:amalykhi-High-OCP-65071 @OCP-65071 @amalykhi", func() {
+			It("Author:amalykhi-High-OCP-65071 subnet_id option is available for machinepool for BYO VPC single-az cluster", ci.Day2, ci.High, ci.FeatureMachinepool, func() {
+				if profile.MultiAZ || !profile.BYOVPC {
+					Skip("The test is configured for SingleAZ BYO VPC cluster only")
+				}
+				getResp, err := cms.RetrieveClusterDetail(ci.RHCSConnection, clusterID)
+				var zones []string
+				vpcOutput, err := ci.PrepareVPC(profile.Region, false, true, zones, getResp.Body().Name())
+
+				By("Tag new subnet to be abbly apply it to the machinepool")
```
Comment (shellymiron) [inline tests/e2e/machine_pool_test.go:440]:
```suggestion
				By("Tag new subnet to be able to apply it to the machinepool")
```

File: tests/e2e/machine_pool_test.go [hunk 2]
```diff
@@ -426,5 +428,61 @@ var _ = Describe("TF Test", func() {
...
+				tagValue := "shared"
+				VPCTagArgs := &exe.VPCTagArgs{
+					AWSRegion: getResp.Body().Region().DisplayName(),
+					IDs:       vpcOutput.ClusterPrivateSubnets,
+					TagKey:    tagKey,
+					TagValue:  tagValue,
+				}
+				err = VpcTagService.Create(VPCTagArgs)
+				Expect(err).ToNot(HaveOccurred())
+				By("Create additional machinepool with subnet id specified")
```
Comment (shellymiron) [inline tests/e2e/machine_pool_test.go:452]:
```suggestion
				
				By("Create additional machinepool with subnet id specified")
```

File: tests/e2e/machine_pool_test.go [hunk 3]
```diff
@@ -426,5 +428,61 @@ var _ = Describe("TF Test", func() {
...
+				VpcTagService := exe.NewVPCTagService()
+				tagKey := fmt.Sprintf("kubernetes.io/cluster/%s", getResp.Body().InfraID())
+				tagValue := "shared"
+				VPCTagArgs := &exe.VPCTagArgs{
+					AWSRegion: getResp.Body().Region().DisplayName(),
+					IDs:       vpcOutput.ClusterPrivateSubnets,
+					TagKey:    tagKey,
+					TagValue:  tagValue,
+				}
+				err = VpcTagService.Create(VPCTagArgs)
```
Comment (shellymiron) [inline tests/e2e/machine_pool_test.go:450]: do we want to also verify the tags are indeed added to the subnet?

Comment (amalykhi) [reply tests/e2e/machine_pool_test.go:450]: not sure, it is not the subject of the testing,  it is actually done for the cluster to allow applying the subnet to the machinepool of the cluster.

**Pull request merged**